Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WebKit.h"
+
+WK_EXTERN NSErrorDomain const _WKWebExtensionErrorDomain = WKWebExtensionErrorDomain;
+WK_EXTERN NSNotificationName const _WKWebExtensionErrorsWereUpdatedNotification = WKWebExtensionErrorsWereUpdatedNotification;
+
+WK_EXTERN
+@interface _WKWebExtension : WKWebExtension
```

Comment:
Just for my own edification -- Is there any particular reason to use an empty inheriting interface to define these symbols rather than `@compatibility_alias`?

Reply:
Great catch! I think that should work as well (and would be a lot cleaner, assuming it does).

Reply:
After a cursory Slack search it seems like this might just affect headers, and not what symbols are actually exported during build -- so not sure if this will work for binary compatibility purposes.

Reply:
I see, thanks for double checking!

Reply:
I could have used that instead of `#define` though! I'll keep that in mind for next time.